### PR TITLE
first() and last() can be called without dplyr loaded. closes #3498

### LINKED ISSUE #3498: Non-hybrid first() and last() fail if dplyr is not attached
``` r
library(magrittr)
tibble::tibble(a = 1:3) %>% dplyr::summarise(a = dplyr::first(.data$a))
#> Error in summarise_impl(.data, dots): Evaluation error: could not find function "nth".
```

> Created on 2018-04-11 by the [reprex
package](http://reprex.tidyverse.org) (v0.2.0).

I think this is due to how `firstlast_prototype()` is implemented. Several downstream packages fail because of this.

The other question is why hybrid doesn't work here. Would a handler for `$` that looks for a `.data` LHS be sufficient here?

@romainfrancois: Could you please take a look? I'll rerun revdepchecks when this is fixed.

### PR DESCRIPTION
**Pull request by romainfrancois**

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -48,7 +48,8 @@ Suggests:
     RPostgreSQL (>= 0.6.2),
     RSQLite (>= 2.0),
     testthat (>= 2.0.0),
-    withr (>= 2.1.1)
+    withr (>= 2.1.1),
+    callr
```
Comment (krlmlr) [inline DESCRIPTION]: `usethis::use_tidy_description()` takes care of maintaining the order.

File: src/hybrid_nth.cpp [hunk 2]
```diff
@@ -218,6 +218,9 @@ Result* nth_prototype(SEXP call, const ILazySubsets& subsets, int nargs) {
     return 0;
   }
   SEXP data = maybe_rhs(CADR(call));
+  if (TYPEOF(data) == LANGSXP && CAR(data) == R_DollarSymbol && CADR(data) == Rf_install(".data") && TYPEOF(CADDR(data)) == SYMSXP ){
```
Comment (krlmlr) [inline src/hybrid_nth.cpp]: I think supporting `.data$` is a bigger issue better solved elsewhere, as part of the all-or-nothing hybridization we discussed yesterday.

**Comment by romainfrancois** [comment]
I'll try something different.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**